src/file.js

```diff
@@ -153,6 +153,25 @@
   };
 
   /**
+   * Method to replace an extension, if one does not exist in the file string, it will be added.
+   *
+   * @param {string} fileString - File string to add the extension to if one does not exist
+   * @param {string} extension - Extension to add if one does not exist in `fileString`. The
+   *   value is the extension without the `.`. E.g. `js`, `html`.  Not `.js`, `.html`.
+   * @returns {string} New fileString with the new extension if one did not exist
+   */
+  File.replaceExtension = function(fileString, extension) {
+    var fileName  = File.parseFileName(fileString),
```

Comment:
I had to change the capture group to the beginning of the regex, like so: `([^.\/\\]+\.)[^.]+$` and use the `$1` variable to retrieve the result of the capture group to be used within the regex.  3f8c74a seems to fix all of the problems I've been having with the code so far. It does still include an if statement to check for the case when no extension is present in the fileString, at which point it simply concatenates the extension onto the filename.